lib/frontend/src/frontend.rs

```diff
@@ -591,14 +638,37 @@ impl<'a> FunctionBuilder<'a> {
         });
 
         let ch = self.ins().uextend(types::I32, ch);
-        self.ins().call(libc_memset, &[buffer, ch, len]);
+        self.ins().call(libc_memset, &[buffer, ch, size]);
+    }
+
+    /// Calls libc.memset
+    ///
+    /// Writes `size` bytes of value `ch` to memory starting at `buffer`.
+    pub fn emit_small_memset(&mut self,
+                             isa: &TargetIsa,
+                             buffer: Value,
+                             ch: Value,
```

Comment:
I don't think I mentioned this yet, but for the small memset function, we actually want the `ch` to be passed in as a constant too, since the main use for this function is just zeroing out memory.

Also, we want this function to take an alignment parameter similar to memcpy for its destination pointer, and we can compute an access size from it in a similar way. And in that case we'll need to convert `ch` from a single byte to an access-size-sized integer in which every byte is `ch` :).